tests/functionality_tests/test_activation_shapes.py

```diff
@@ -35,7 +42,7 @@ def main():
             rec_row_start, rec_col_start, rec_row_limit, rec_col_limit
         )
         grid.find_border_nodes()
-        way_points = grid.get_all_guided_lawnmower_waypoints_adjustable(is_vertical)
+        way_points = grid.get_all_guided_lawnmower_waypoints_adjustable(direction)
```

Comment:
grid.find_border_node() and waypoints = ... can be moved outside of if statements to reduce duplicate code